age/smart-contract/assembly/contracts/main.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * This file is a smart contract that enables you to store and retrieve the age of a person by their name.
+ *
+ * @remarks
+ * The smart contract is written in AssemblyScript and executed on the Massa blockchain.
+ *
+ * The contract has three functions:
+ *
+ * - The constructor sets up the storage by creating an initial entry for a person with a default age value.
+ *  This function is always called once on contract deployment.
+ *
+ * - The change_age function allows you to update a person's age by providing their name and a new age value.
+ *  If the entry for the given name does not exist, the person is created.
+ *  This function generates an event that indicates the changes that are made.
+ *
+ * - The get_age function retrieves the age of a person by their name.
+ *  If the entry for the given name does not exist, the execution is aborted.
+ *  This function returns the serialized 'age' found, and generates an event that indicates the retrieval.
+ *
+ * The smart contract interacts with the Massa blockchain using the massa-as-sdk,
+ * which provides functions for storage, access control, event generation and more.
+ * @see [massa-as-sdk](https://github.com/massalabs/massa-as-sdk)
+ *
+ */
+import { Args } from '@massalabs/as-types';
+import {
+  Storage,
+  callerHasWriteAccess,
+  generateEvent,
+} from '@massalabs/massa-as-sdk';
+
+/**
+ * This function is the constructor, it is always called once on contract deployment.
+ * In this case, the constructor is adding a person
+ * named 'alice' with an age of 1 as default first entry in the Storage.
+ *
+ * @param args - The serialized arguments (unused).
+ *
+ * @returns none
+ *
+ */
+export function constructor(_: StaticArray<u8>): StaticArray<u8> {
```

Comment:
Isn't any sc function forced to have 'StaticArray<u8>' return type for any kind of interfacing ?

Reply:
No, it's more complex that that.
If i remember well, you can return:
- void
- stack only object
- StaticArray<u8>

But it would be cool to have a confirmation of the core team on this.